src/ansys/sound/core/sound_composer/source_spectrum.py

```diff
@@ -0,0 +1,238 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound Composer's spectrum source."""
+import warnings
+
+from ansys.dpf.core import Field, Operator
+from matplotlib import pyplot as plt
+import numpy as np
+from numpy import typing as npt
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+from ._source_parent import SourceParent
+from .source_control_spectrum import SourceControlSpectrum
+
+ID_COMPUTE_LOAD_SOURCE_SPECTRUM = "sound_composer_load_source_spectrum"
+ID_COMPUTE_GENERATE_SOUND_SPECTRUM = "sound_composer_generate_sound_spectrum"
+
+
+class SourceSpectrum(SourceParent):
+    """Sound Composer's spectrum source class.
+
+    This class creates a spectrum source for the Sound Composer. A spectrum source is used to
+    generate a sound signal from a given spectrum and its source control. The source control
+    contains the duration of the sound and the method used to generate it.
+    """
+
+    def __init__(self, file_source: str = "", source_control: SourceControlSpectrum = None):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file_source : str, default ""
+            Path to the spectrum file.
+        source_control : SourceControlSpectrum, default None
+            Source control to use when generating the sound from this source.
+        """
+        super().__init__()
+        self.source_control = source_control
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_COMPUTE_LOAD_SOURCE_SPECTRUM)
+        self.__operator_generate = Operator(ID_COMPUTE_GENERATE_SOUND_SPECTRUM)
+
+        # load_source_spectrum can only be called after __operator_load is defined.
+        if len(file_source) > 0:
+            self.load_source_spectrum(file_source)
+        else:
+            self.source_spectrum_data = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.source_spectrum_data is not None:
+            support_data = self.source_spectrum_data.time_freq_support.time_frequencies.data
+            str_source = (
+                f"'{self.source_spectrum_data.name}'\n"
+                f"\tFmax: {support_data[-1]:.0f} Hz\n"
+                f"\tDeltaF: {support_data[1] - support_data[0]:.1f} Hz"
```

Comment:
Here, we need a check that 
support_data length is >= 1 to ensure that support_data[0] exists
then that
support_data length is >= 2 to ensure that support_data[1] exists

Reply:
@anshlachamb I added this check in the setter rather than here

Reply:
As discussed, the setter check only checks if > 0, and then the str is adjusted in case there is only 1 point ("N/A" is displayed)